From 786c6183c040ed6f6e440410957aa42d4086df03 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 19 Oct 2023 00:35:11 +0800
Subject: [PATCH 47/89] mtk: mac80211: increase association timeout time

Prevent from sending multiple association requests while AP is already
hanlding the request.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 net/mac80211/mlme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -7792,7 +7792,7 @@ void ieee80211_sta_work(struct ieee80211
 			 */
 			if (status_acked) {
 				ifmgd->assoc_data->timeout =
-					jiffies + IEEE80211_ASSOC_TIMEOUT_SHORT;
+					jiffies + IEEE80211_ASSOC_TIMEOUT_SHORT * 4;
 				run_again(sdata, ifmgd->assoc_data->timeout);
 			} else {
 				ifmgd->assoc_data->timeout = jiffies - 1;
